proto/kvrpcpb.proto

```diff
@@ -173,6 +178,29 @@ message PessimisticLockRequest {
     bool check_existence = 12;
     // TiKV lock the record only when it exists
     bool lock_only_if_exists = 13;
+    PessimisticLockWaitingMode lock_waiting_mode = 14;
+}
+
+enum PessimisticLockKeyResultType {
+    Empty = 0;
+    Value = 1;
+    Existence = 2;
+    LockedWithConflict = 3;
+    Failed = 4;
+}
+
+message PessimisticLockKeyResult {
+    PessimisticLockKeyResultType type = 1;
+    bytes value = 2;
+    bool existence = 3;
+    // We allow a key be locked when there is write conflict (latest commit_ts > for_update_ts).
+    // In this case, the key is semantically locked by a newer for_update_ts.
+    // For each requested key, non-zero if the key is locked with write conflict, and the
+    // value is the commit_ts of the latest version of the specified key, which is the same value
+    // as the actual for_update_ts. At the sametime, the corresponsing value in `values` field and
+    // `not_founds` field will be the latest value of the key even if `return_values` and
+    // `check_existence` are not set.
+    uint64 locked_with_conflict_ts = 4;
```

Comment:
> is the commit_ts of the latest version of the specified key, which is the same value as the actual for_update_ts.

I'm confused about this sentence. If the key is locked with conflict and the `commit_ts` of the key is larger than the try-lock transaction `for_update_ts`, the `locked_with_conflict_ts` would be set to the latest `commit_ts`?

Reply:
Yes. I adjusted the expression of the comment.